client/rpc/server.go

```diff
@@ -0,0 +1,124 @@
+package rpc
+
+import (
+	"encoding/json"
+	"math/big"
+
+	"github.com/nats-io/nats-server/v2/server"
+	"github.com/nats-io/nats.go"
+	"github.com/rs/zerolog"
+	nitro "github.com/statechannels/go-nitro/client"
+	"github.com/statechannels/go-nitro/network"
+	"github.com/statechannels/go-nitro/network/serde"
+	natstrans "github.com/statechannels/go-nitro/network/transport/nats"
+	"github.com/statechannels/go-nitro/protocols/directdefund"
+	"github.com/statechannels/go-nitro/protocols/directfund"
+)
+
+// RpcServer handles nitro rpc requests and executes them on the nitro client
+type RpcServer struct {
+	nts     *network.NetworkService
+	ns      *server.Server
```

Comment:
Just flagging the tight coupling to `nats-server` here.

Reply:
Good catch. Adding a second transport should be helpful in flushing out transport specific code like this.